### ISSUE TITLE
When running MultipleClassFileToPsr4ClassesRector on multiple files: "Return value of PhpParser\Internal\TokenStream::getIndentationBefore()          must be of the type int, null returned".

### ISSUE THREAD

**Issue by JanMikes** [rectorphp/rector#1408]
Command:
```
vendor/bin/rector process --config config/rector/psr-4-classes-split.yaml src
```

Config:
```yaml
services:
    Rector\Rector\Psr4\MultipleClassFileToPsr4ClassesRector: ~

parameters:
    exclude_paths:
        - '**/template.php'
```

*When i run it separately file by file, it works.*

Here is call stack (when running with -vvv):
```log
Fatal error: Uncaught TypeError: Return value of PhpParser\Internal\TokenStream::getIndentationBefore() must be of the type int, null returned in /project/vendor/nikic/php-parser/lib/PhpParser/Internal/TokenStream.php:192
Stack trace:
#0 /rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(763): PhpParser\Internal\TokenStream->getIndentationBefore(17)
#1 /rector/src/PhpParser/Printer/BetterStandardPrinter.php(87): PhpParser\PrettyPrinterAbstract->pArray(Array, Array, 12, 0, 'stmts', NULL, '')
#2 /rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(573): Rector\PhpParser\Printer\BetterStandardPrinter->pArray(Array, Array, 12, 0, 'stmts', NULL, '\n')
#3 /rector/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php(867): PhpParser\PrettyPrinterAbstract->p(Object(PhpParser\Node\Stmt\Class_), true)
#4 /rector/src/PhpParser/Printer/BetterStandardPrinter.php(87): PhpParser\PrettyPrinterAbstract->pArray(Array, Array, 5, 0, 'stmts', NULL, '')
#5 /rector/vendor/nikic/php-parser/lib/Ph in /project/vendor/nikic/php-parser/lib/PhpParser/Internal/TokenStream.php on line 192
```

I noticed it stops processing on file ARES.php which contains 3 classes. But when i run it only on this file it splits it correctly.

In `/src` there are 600 files and i got following errors when running on whole directory:

```log
~/Sites/klarka  jm/psr-4-libs-to-src-2-split-classes ✗                                                                                                                                                                            13m ⚑  ⍉
▶ docker run -v $(pwd):/project gitlab.devklarka.cz:5000/infrastructure/web-development/rector:latest process --autoload-file /project/vendor/autoload.php --config /project/config/rector/psr-4-classes-split.yaml /project/src
Rector No version set (parsed as 1.0.0)@
Config file: /project/config/rector/psr-4-classes-split.yaml

   0/588 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
  58/588 [▓▓░░░░░░░░░░░░░░░░░░░░░░░░░░]   9%
 116/588 [▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░]  19%
 174/588 [▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░]  29%
 232/588 [▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░]  39%
 290/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░]  49%
 348/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░]  59%
 406/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░]  69%
 464/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░]  78%
 522/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░]  88%
 580/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░]  98%
 588/588 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 [ERROR] Could not process "/project/src/ARES.php" file, due to:                
         "Return value of PhpParser\Internal\TokenStream::getIndentationBefore()
         must be of the type int, null returned".                                      
```

**Comment by TomasVotruba** [comment]
Please make it short and less verbose. 1 error is enough

Do version of php-parser match?

If so, add test case to `MultipleClassFileToPsr4ClassesRectorTest` with code that causes this. E.g.  `Ares.php`

**Comment by JanMikes** [comment]
Yeah, i have edited original message.

Versions of php-parser matches - 4.2.1

Sure, i will try to add minimal required code to break things up 😄

**Comment by JanMikes** [comment]
@TomasVotruba i have found out that problem is only when processing multiple than one files.

Was the rector originally supposed to process multiple files (i run it on directory).

At the moment `MultipleClassFileToPsr4ClassesRectorTest.php` is not ready to process directories, i have tried something, but unfortunately not successfully - i was trying to detect whether target is directory/file and if it is directory, register robotloader on it so all classes inside are autoloaded.

Any idea?

For now i am able to workaround that running on each file separately (i will write script that will run it, because running manually it on 300+ files is insane 😄 )

**Comment by JanMikes** [comment]
I managed to create minimal demo to repro (in branch `psr-4-bug-demo`):

Before: check `MultipleFilesPsr4Bug/Classes.php`

- `git checkout psr-4-bug-demo`
- `bin/rector process MultipleFilesPsr4Bug --config MultipleFilesPsr4Bug/config.yaml --debug`

Then go and check file `MultipleFiles/ClassOne.php` and `MultipleFiles/ClassTwo.php` they are completely malformed.

**Comment by TomasVotruba** [comment]
I didn't notice reply. Good job with demo!

I'll check it today

**Comment by TomasVotruba** [comment]
Got the bug locally, I checked it for 20 minutes with no success and it looks really nice :D
I suspect some cache, I'll keep digging

**Comment by TomasVotruba** [comment]
Resolved in #1464
